automation/tests-storage.sh

```diff
@@ -12,28 +33,107 @@ create_loop_devices() {
 }
 
 setup_storage() {
-    python3 tests/storage/userstorage.py setup
+    # Configure lvm to ignore udev events, otherwise some lvm tests hang.
+    mkdir -p /etc/lvm
+    cp docker/lvmlocal.conf /etc/lvm/
+
+    # Make sure we have enough loop device nodes. Using 16 devices since with 8
+    # devices we have random mount failures.
+    create_loop_devices 16
+
+    # Build vdsm.
+    ./autogen.sh --system
+    make
+
+    # Setup user stoage during the tests.
+    make storage
 }
 
 teardown_storage() {
-    python3 tests/storage/userstorage.py teardown \
+    make clean-storage \
         || echo "WARNING: Ingoring error while tearing down user storage"
 }
 
-# Configure lvm to ignore udev events, otherwise some lvm tests hang.
-mkdir -p /etc/lvm
-cp docker/lvmlocal.conf /etc/lvm/
+# Process command-line arguments
+while test $# -gt 0
+    do
+        case $1 in
 
-# Make sure we have enough loop device nodes. Using 16 devices since with 8
-# devices we have random mount failures.
-create_loop_devices 16
+            root )
 
-# Build vdsm.
-./autogen.sh --system
-make
+                mode=root
+                ;;
+            
+            vdsm )
+
+                mode=vdsm
+                ;;
+
+            -u=* | --user=* )
+
+                user="${1#*=}"
+                if [ ! -z ${mode+x} ]; then
+                    mode=vdsm
+                fi
+                ;;
+            
+            -u | --user )
+
+                shift
+                if [ -z $1 ]; then
+                    echo "error: user expected after --user"
+                    exit 1
+                fi
+                user="$1"
+                if [ ! -z ${mode+x} ]; then
+                    mode=vdsm
+                fi
+                ;;
+            
+            * )
+
+                echo "Unknown option: $1"
+                print_help
+                exit 1
+            ;;
+
+        esac
+        shift
+done
+
+# Defaults if not provided
+if [ ! -z ${user+x} ]; then
+    user=root
+fi
+if [ ! -z ${mode+x} ]; then
+    mode=root
+fi
+
+if [ "$user" != "root" -a "$mode" = "root" ]; then
+    echo "error: non-privileged user ($user) can not run tests marked as root"
+    print_help
+    exit 1
+fi
+
+echo "Running $mode tests as user $user"
 
-# Setup user stoage during the tests.
-trap teardown_storage EXIT
 setup_storage
+if [ "$user" != "root" ]; then
+    # Change ownership of current and storage folders
+    # to allow non-privileged access.
+    chown -R vdsm:kvm ./ /var/tmp/vdsm*
+fi
```

Comment:
Keeping both user and mode seems too complicated. Can we use only
user variable?